### PR TITLE
Scope de-elevation

### PR DESCRIPTION
**Pull request by jtracey93**
# Overview/Summary

As per https://github.com/Azure/bicep-registry-modules/issues/394, this PR is de-elevating the scopes access is required to and no longer creating a tenant level deployment, which requires `/` RBAC access.

We also add some testing for releases

## This PR fixes/adds/changes/removes

1. Remove `/` scope access requirements and replace with `managementGroup()`
2. Add testing for releases

### Breaking Changes

Not technically breaking, but a cleanup of permissions could be done

## Testing Evidence

Tests will suffice

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/bicep-lz-vending/wiki/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/bicep-lz-vending/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/bicep-lz-vending/issues)
- [x] *(LZ-Vending Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/lz-vending/backlog)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/bicep-lz-vending/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/bicep-lz-vending/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/bicep-lz-vending/tree/main/.github/workflows/code-review.yml)
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)

### CODE REVIEW

File: tests/pester/release.tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  Context "version.json file tests" {
+
+    BeforeAll {
+      $versionFile = Get-Content $versionFilePath -Raw | ConvertFrom-Json
+      $gitRepoLatestTag = git describe --tags --abbrev=0
+      $releaseNotesUrlSplitLast = $versionFile.releaseNotes.Split("/")[-1]
+      $releaseNotesUrlStart = "https://github.com/Azure/bicep-lz-vending/releases/tag/v"
+
+      # Download the previous version.json file from the repo
+      $previousVersionRawUrl = "https://raw.githubusercontent.com/Azure/bicep-lz-vending/v1.2.2/version.json"
```
Comment (matt-FFFFFF) [inline tests/pester/release.tests.ps1:19]: The tag here is hard coded - is this intentional? We can use `$gitRepoLatestTag` instead?

File: tests/pester/release.tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,58 @@
+[CmdletBinding()]
+param (
+  [Parameter(Mandatory = $false)]
+  [string]
+  $versionFilePath = "./version.json"
```
Comment (matt-FFFFFF) [inline tests/pester/release.tests.ps1:5]: Consider combining version and gitTag in this file. They are the same save for a v prefix. This is simple to do in code and removes the need to update two values that could get out of sync.

**Review by matt-FFFFFF — COMMENTED** [review]
Couple of comments - not requesting changes but rather some things to consider

File: tests/pester/release.tests.ps1 (hunk 1 above)
Comment (jtracey93) [reply tests/pester/release.tests.ps1:19]: Good spot left over from testing. Fixed

File: tests/pester/release.tests.ps1 (hunk 2 above)
Comment (jtracey93) [reply tests/pester/release.tests.ps1:5]: will remove gitTag

**Review by matt-FFFFFF — APPROVED** [review]

**Pull request merged**